ChangeLog.d/config_psa-md_light.txt

```diff
@@ -0,0 +1,9 @@
+Bugfix
+   * Fix a compilation error in some configurations with
+     MBEDTLS_PSA_CRYPTO_CONFIG and MBEDTLS_USE_PSA_CRYPTO enabled but
+     MBEDTLS_MD_C not explicitly enabled.
```

Comment:
I'm bit of confused of this Bugfix. `MBEDTLS_MD_C` is enabled either by `MBEDTLS_PKCS5_C` or by `PSA_WANT_ALG_DETERMINISTIC_ECDSA`. How does the compilation error happen and how is fixed? Could you please provide more detail here?

Reply:
I haven't actually tested yet. Once I get the time I might add it as a non-regression test.

We recently noticed a problem in some builds with `MBEDTLS_ECP_LIGHT` not getting enabled because the ECP_C→ECP_LIGHT activation came before the PSA_xxx→ECP_C activation. ECP light is not released yet so fixing this is not a change from the previous release. From reading the source code, I think there's a similar problem with RSA_C→MD_LIGHT and PSA_xxx→RSA_C. This has been around since MD light was introduced, so it is a bug in a released version.

Reply:
Right, there's similar problem with RSA_C→MD_LIGHT. But the difference between ECP_C→ECP_LIGHT is that we have multiple ways to enable `MBEDTLS_MD_LIGHT`. It's even harder to spot the error.

I think I understand what the bug is in this Bugfix. Should we change `enabled but
     MBEDTLS_MD_C not explicitly enabled.` as `enabled but
     MBEDTLS_MD_LIGHT not explicitly enabled.` or something else? Since the bug is `MD_LIGHT` is not enabled correctly due to wrong inclusion order of `config_psa.h`. But `MBEDTLS_MD_C` is enabled as expected in `config_psa.h` by `PSA_XXX`.

Reply:
`MBEDTLS_MD_LIGHT` is not a public option. It gets enabled when either `MBEDTLS_MD_C` is enabled or one of the legacy crypto modules that require it is enabled (e.g. `MBEDTLS_RSA_C`). Until this PR, if `RSA_C` is enabled indirectly via a `PSA_WANT` and `MD_C`, `MD_LIGHT` doesn't get enabled (because `MD_C` and `RSA_C` get enabled after the code that might enable `MD_LIGHT`) and this can cause the build to fail. From the point of view of the public interface, the problem is having enabled (e.g.) `PSA_WANT_ALG_RSA_PSS` but not `MBEDTLS_MD_C`.